### DISCUSSION TITLE
Auth scope of scheduled jobs

### DISCUSSION THREAD

**Discussion by thezachdrake** [pocketbase/pocketbase#5226] [Q&A] [answered]
https://pocketbase.io/docs/go-jobs-scheduling/

I think I have figured this out but the docs don't say specifically so I would appreciate someone confirming. Do scheduled tasks get triggered as ADMIN auth scope? With API routes (https://pocketbase.io/docs/go-routing/) there is auth information stored in the context store. However, scheduled tasks to do not have that context store passed to them.

**Comment by ganigeorgiev** [comment]
I'm not entirely sure that I understand the question.

The jobs scheduling doesn't have information about any auth state and they are not triggered by HTTP requests. The scheduled jobs are simple functions that are called repeatedly based on the specified cron expression (aka. similar to the OS cron command).
It is up to the developer what operations that function/job will do (ex. access or modify the database, send email(s), generate reports, etc.).

**Reply by ganigeorgiev** [reply]
Maybe the confusion comes from because in the docs the `cron` instance is created as part of the `OnBeforeServe` hook? If that's the case this is because we want the cron to run only when the long-running app process is active (aka. when the executable was started with `./pocketbase serve`) and not when a one-off command was invoked (eg. `./pocketbase admin update ...`).

With the [refactoring](https://github.com/pocketbase/pocketbase/discussions/4355) jobs scheduling is simplified and there is no longer need to manually create a scheduler instance and can simply call `app.Cron().MustAdd(...)` (_eventually in the future this will enable us also to show the registered cron jobs in the Admin UI to trigger them manually with a button_).

**Reply by thezachdrake** [reply]
More concrete example would help. I am creating a scheduled job that pulls stored info from my users, then reaches out to an external api and gets the data that matches them, and then pulls that back in and writes into a collection. I have permissions on the user data to make only their data available, and I have the same on the collection that this job populates. So what I want to make sure is that if I schedule a job and it makes a DB query, is that query run with admin privileges (i.e. can access all records) or will I need to control for access permissions.

**Reply by ganigeorgiev** [reply]
It is not clear to me what do you mean and I still don't understand from where the confusion comes from but there is no auth scope or privileges when running a scheduled job (why and from where it would come? it doesn't make sense to me).

If you executing a DB query with one of the [raw DB](https://pocketbase.io/docs/go-database/) or [Record](https://pocketbase.io/docs/go-records/) helpers then you are directly communicating with the database. There are no "admin" or any other privileges here.

**Reply by ganigeorgiev** [reply, marked as answer]
If by _"I have permissions on the user data"_ you mean the collection API rules then these rules apply **only** when operating with the [Web APIs](https://pocketbase.io/docs/api-records/).
When extending PocketBase with Go, or the JSVM, you have direct access to the database and are in full control.

**Reply by thezachdrake** [reply]
That's what I needed to understand! So anything I write myself has unfiltered access to the database unless I choose to put access on it by checking the auth in the context(https://pocketbase.io/docs/go-routing/) and that context only exists when receiving requests from the echo router.
